mesonbuild/backend/backends.py

```diff
@@ -61,11 +62,11 @@ class TargetIntrospectionData(TypedDict):
 # Assembly files cannot be unitified and neither can LLVM IR files
 LANGS_CANT_UNITY = ('d', 'fortran', 'vala')
 
+@dataclass(eq=False)
 class RegenInfo:
-    def __init__(self, source_dir: str, build_dir: str, depfiles: T.List[str]):
-        self.source_dir = source_dir
-        self.build_dir = build_dir
-        self.depfiles = depfiles
+    source_dir: str
+    build_dir: str
+    depfiles: T.List[str]
```

Comment:
with Python >= 3.7, if the file starts with

```python
from __future__ import annotations
```

and then annotation like:

```python
depfiles: list[str]
```

Reply:
We could do that on 3.6 too. Future annotations just means we don't have to quote it.

It would then... fail with mypy on older versions of python before list became a generic type? I dunno. Does mypy explicitly handle that or just rely on python to evaluate the types. I'd assume the latter...

I guess it's not too terrible if lint_mypy requires newer python than meson itself.

Reply:
It requires a pretty up to date mypy, which may or may not be nice. I'd rather make that change as a separate, explicit change anyway

Reply:
On the other hand, this PR doesn't actually make use of future annotations. :D

I do have some preliminary work on slogging through that... https://github.com/eli-schwartz/meson/commits/python37-additions